backend/src/main/java/org/cryptomator/hub/license/LicenseHolder.java

```diff
@@ -75,6 +85,45 @@ public void set(String token) throws JWTVerificationException {
 		settings.persist();
 	}
 
+	/**
+	 * Attempts to refresh the Hub licence every day at 01:00 UTC if claim refreshURL is present.
+	 */
+	@Scheduled(cron = "0 1 * * * ?", timeZone = "UTC")
+	void refreshLicenseScheduler() throws InterruptedException {
+		if (license != null) {
+			var refreshUrl = licenseValidator.refreshUrl(license.getToken());
+			if (refreshUrl.isPresent()) {
+				var client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
+				refreshLicense(refreshUrl.get(), license.getToken(), client);
+			}
+		}
+	}
+
+	//visible for testing
+	void refreshLicense(String refreshUrl, String license, HttpClient client) throws InterruptedException {
+		var parameters = new HashMap<String, String>();
+		parameters.put("token", license);
+		var body = parameters.entrySet() //
+				.stream() //
+				.map(e -> e.getKey() + "=" + URLEncoder.encode(e.getValue(), StandardCharsets.UTF_8)) //
+				.collect(Collectors.joining("&"));
+		var request = HttpRequest.newBuilder() //
+				.uri(URI.create(refreshUrl)) //
+				.headers("Content-Type", "application/x-www-form-urlencoded") //
+				.POST(HttpRequest.BodyPublishers.ofString(body))  //
```

Comment:
Isn't there a body publisher for form-urlencode that handles all the encoding for us?

Reply:
I researched for a while but couldn't find one...

Reply:
Apparently, there isn't. This [isn't the first time](https://github.com/coffeelibs/tiny-oauth2-client/blob/1bcba9b4e755fd7b97585c3375e2074994f5e864/src/main/java/io/github/coffeelibs/tinyoauth2client/TinyOAuth2Client.java#L183) I looked it up, I feel like I should add a feature request to the JDK.